kong/tools/string.lua

```diff
@@ -24,16 +31,49 @@ _M.split = pl_stringx.split
 
 --- strips whitespace from a string.
 -- @function strip
-_M.strip = function(str)
-  if str == nil then
+_M.strip = function(value)
+  if value == nil then
     return ""
   end
-  str = tostring(str)
-  if #str > 200 then
-    return str:gsub("^%s+", ""):reverse():gsub("^%s+", ""):reverse()
-  else
-    return str:match("^%s*(.-)%s*$")
+
+  if type(value) ~= "string" then
+    value = tostring(value)
+  end
```

Comment:
I have a question about this. `tostring` may return nil value if it's unable to convert according to [lua_tolstring](https://github.com/lua/lua/blob/master/lapi.c#L423) and cause panic. And is it good to assume `to_string` should never return a string with space? I personally feel strange that `strip` can operate on a non-string type?

Reply:
@git-hulk, we csn add check for `nil`. I was actually thinking about it. Operating with non-string, comes from original implementation. I dodn’t want to have a behavioral change with this, but I will add a comment about it.

Reply:
I see, thanks for your prompt reply.

Reply:
Check added.